src/main/java/de/tum/in/www1/artemis/service/scheduled/NotificationScheduleService.java

```diff
@@ -164,8 +164,8 @@ private void scheduleNotificationForAssessedExercisesSubmissions(Exercise exerci
      * @return true if the time is valid else false
      */
     private boolean checkIfTimeIsCorrectForScheduledTask(ZonedDateTime relevantTime) {
-        // only send a notification if relevantTime is defined and not in the future (i.e. in the range [now-2 minutes, now]) (due to possible delays in scheduling)
-        return relevantTime != null && !relevantTime.isBefore(ZonedDateTime.now().minusMinutes(2)) && !relevantTime.isAfter(ZonedDateTime.now());
+        // only send a notification if relevantTime is defined and not in the future (i.e. in the range [now-2 minutes, now+2 minutes]) (due to possible delays in scheduling)
+        return relevantTime != null && !relevantTime.isBefore(ZonedDateTime.now().minusMinutes(2)) && !relevantTime.isAfter(ZonedDateTime.now().plusMinutes(2));
```

Comment:
Why is this change necessary? Also the comment mentioning "not in the future" and "now + 2 minutes" does not make sense to me.

Reply:
Maybe we should change the comment from "not in the future" to "not in the far future"?